front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { SectionField } from 'components/admin/Section';
+import { useTheme } from 'styled-components';
+import {
+  Box,
+  IOption,
+  ColorPickerInput,
+  Label,
+  Select,
+} from '@citizenlab/cl2-component-library';
+import HeaderImageDropzone from './HeaderImageDropzone';
+import { UploadFile } from 'typings';
+import { ISubmitState } from 'components/admin/SubmitWrapper';
+import { debounce } from 'lodash-es';
+
+// i18n
+import { InjectedIntlProps } from 'react-intl';
+import { injectIntl, FormattedMessage } from 'utils/cl-intl';
+import messages from './messages';
+
+import { isNil, isNilOrError } from 'utils/helperUtils';
+import { convertUrlToUploadFile } from 'utils/fileUtils';
+
+import { ICustomPageAttributes } from 'services/customPages';
+import { IHomepageSettingsAttributes } from 'services/homepageSettings';
+
+import RangeInput from 'components/UI/RangeInput';
+export type PreviewDevice = 'mobile' | 'tablet' | 'desktop';
+
+interface Props {
+  onAddImage: (newImageBase64: string) => void;
+  onRemoveImage: () => void;
+  onOverlayColorChange: (color: string) => void;
+  onOverlayOpacityChange: (color: number) => void;
+  bannerOverlayColor:
+    | IHomepageSettingsAttributes['banner_signed_out_header_overlay_color']
+    | ICustomPageAttributes['banner_overlay_color'];
+  bannerOverlayOpacity:
+    | IHomepageSettingsAttributes['banner_signed_out_header_overlay_opacity']
+    | ICustomPageAttributes['banner_overlay_opacity'];
+  bannerLayout:
+    | IHomepageSettingsAttributes['banner_layout']
+    | ICustomPageAttributes['banner_layout'];
+  headerBg:
+    | IHomepageSettingsAttributes['header_bg']
+    | ICustomPageAttributes['header_bg'];
+  setFormStatus: (submitState: ISubmitState) => void;
+}
+
+const BannerImageField = ({
```

Comment:
Should normally be just a copy and paste from the place this is coming from (GenericHeroBannerForm)